### ISSUE TITLE
very cool.

### ISSUE THREAD

**Issue by PassByYou888** [SkliarOleksandr/NextPascal#1]
Why not update it?
Continue to develop? I'd like to help you.

Best Regards
by.600585

**Comment by ghost** [comment]
What a beautiful project, can you talk more about it? what are the objectives, etc.

**Comment by SkliarOleksandr** [comment]
Hi, Guys. This is my hobby project, so I don't have a lot of time for it. Feel free to make any changes), Now I am focusing on my DelphiAST project. I want to make some static analyzer for Delphi based on it.

**Comment by SkliarOleksandr** [comment]
Short desc of the project. NextPascal project is a like to Delphi language with some improvements:
- ARC memory model with native fast weak references
- Ternary "iif()" operator like ?: 
- break/continue to loop level 
- improvement for generics 
- intermediate "asm" language like ir/il
- and a lot of other things 
also, this project includes my own VM for testing all features with native bindings

**Comment by vtereshkov** [comment]
Is this compiler self-hosting? It seems that its source is too compact for such a broad scope.

**Comment by SkliarOleksandr** [comment]
Yes, it is self-hosted. But it's just a PoC I think. There are no optimizations, no real HW targets except VM, but yes, this compiler can do a lot :-), and of course, there are a lot of bugs)))

**Comment by VincentGsell** [comment]
Congrats ! Very cool project : Hope you can progress on it !
